invenio_rdm_records/services/community_records/service.py

```diff
@@ -0,0 +1,119 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright (C) 2023 CERN.
+#
+# Invenio-RDM-Records is free software; you can redistribute it and/or modify
+# it under the terms of the MIT License; see LICENSE file for more details.
+
+"""RDM Community Records Service."""
+from invenio_pidstore.errors import PIDDoesNotExistError
+from invenio_records_resources.services import (
+    LinksTemplate,
+    RecordService,
+    ServiceSchemaWrapper,
+)
+from invenio_records_resources.services.errors import PermissionDeniedError
+from invenio_records_resources.services.uow import unit_of_work
+from invenio_search.engine import dsl
+
+from invenio_rdm_records.proxies import current_record_communities_service
+
+
+class CommunityRecordsService(RecordService):
+    """Community records service.
+
+    The record communities service is in charge of managing the records of a given community.
+    """
+
+    @property
+    def community_record_schema(self):
+        """Returns the community schema instance."""
+        return ServiceSchemaWrapper(self, schema=self.config.community_record_schema)
+
+    @property
+    def community_cls(self):
+        """Factory for creating a community class."""
+        return self.config.community_cls
+
+    def read(
+        self, identity, community_id, params=None, search_preference=None, **kwargs
+    ):
+        """Search for records published in the given community."""
+        self.require_permission(identity, "read")
+        self.community_cls.pid.resolve(community_id)  # Ensure community's existence
+
+        params = params or {}
+
+        search_result = self._search(
+            "search",
+            identity,
+            params,
+            search_preference,
+            record_cls=self.record_cls,
+            search_opts=self.config.search,
+            extra_filter=dsl.Q("term", **{"parent.communities.ids": str(community_id)}),
```

Comment:
Be aware of the changes here: https://github.com/inveniosoftware/invenio-rdm-records/pull/1189
You will have conflicts and you will have to integrate such changes.